frontend2/src/components/CurrentUserProvider.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from "react";
+import { type UserPrivate } from "../utils/types";
+import { AuthStateEnum, type AuthState, CurrentUserContext } from "../contexts/CurrentUserContext";
+import { Api } from "../utils/api";
+import { removeApiTokens, doApiTokensExist } from "../utils/cookies";
+
+export const CurrentUserProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [userData, setUserData] = useState<{
+    user?: UserPrivate;
+    authState: AuthState;
+  }>({
+    authState: AuthStateEnum.LOADING,
+  });
+
+  const login = (user: UserPrivate): void => {
+    setUserData({
```

Comment:
Should we get rid of `auth.ts` in [https://github.com/battlecode/galaxy/pull/664](https://github.com/battlecode/galaxy/pull/664)? Generally going to have conflicts with api/auth stuff so will need to figure that out.

Reply:
I think auth.ts is good to keep, and eventually we may even split up our api.ts into more files b/c it's rather big

don't worry abt the merge conflicts - we can merge #664 first, then i'll deal with the merge conflicts here

Reply:
> I think auth.ts is good to keep, and eventually we may even split up our api.ts into more files b/c it's rather big
> 
> don't worry abt the merge conflicts - we can merge #664 first, then i'll deal with the merge conflicts here

Sounds good!